diff --git a/mailchimp.module b/mailchimp.module
index 8bf0a21..61017d0 100644
--- a/mailchimp.module
+++ b/mailchimp.module
@@ -80,7 +80,7 @@ function mailchimp_subscribe_user($list, $email, $merge_vars, $message = TRUE, $
     if ($message && $success && $double_optin) {
       drupal_set_message(t('You have chosen to subscribe to %list. An email 
         will be sent to your address. Click the link in the email to confirm 
-        the subscription.', array('%list' => $list->name)));
+        the subscription.', array('%list' => $list->label)));
     }
     elseif ($message && $success) {
       drupal_set_message(t('You have subscribed to %list.', array('%list' => $list->label)));
diff --git a/modules/mailchimp_lists/mailchimp_lists.module b/modules/mailchimp_lists/mailchimp_lists.module
index dfa7ff2..bda7c1d 100644
--- a/modules/mailchimp_lists/mailchimp_lists.module
+++ b/modules/mailchimp_lists/mailchimp_lists.module
@@ -384,7 +384,7 @@ function mailchimp_lists_auth_newsletter_form(&$form, $list, $account) {
   if ($list->list_type == MAILCHIMP_LISTTYPE_OPTIONAL) {
     $form[$wrapper_key]['subscribe'] = array(
       '#type'           => 'checkbox',
-      '#title'          => ($list->label) ? t($list->label) : t('Subscribe to the @newsletter newsletter', array('@newsletter' => $list->name)),
+      '#title'          => ($list->label) ? t($list->label) : t('Subscribe to the @newsletter newsletter', array('@newsletter' => $list->label)),
       '#default_value'  => $is_subscribed,
       '#description'    => $list->description,
     );
